### ISSUE TITLE
Add visual studio support

### ISSUE THREAD

**Issue by Joren-Thijs** [dotnet/tye#698]
### What should we add or change to make your life better?

It would be great to have a Tye extension available in visual studio. (All i need is a `tye run` button)

### Why is this important to you?

Right now i can't easily transistion to tye because part of my team wants to stay with visual studio and they don't want to run their project from the commandline. It would be nice to have a replacement for the run multiple projects button in VS. Right now if i still want to use Tye i have to manually setup Tye's env variable structure in the launchsettings.json to make sure it will work for them when using the traditional method. If tye could generate this for me then that would be a great help too.

**Comment by tebeco** [comment]
yeah i would love a to see if "just press F5" is possible to use either the `tye.yaml` by default
but also the possibility to select one (mono repo)
and that would attach debugger properly to the given `project` ;)

**Comment by gnaungayan** [comment]
I managed to create a workaround for the meantime to have our team a "F5"experience

1. Install Microsoft Child Process Debugging Power Tool
2. Enable Native Code Debugging on target projects that you wanted to Debug.
3. Create a project that serves as your Runner or Startup project.
4. Create/Edit a debug profile and change it to Executable.
5. On the Executable field, enter **"tye"**(without double quotes) and on the arguments field enter **"run"** (without double quotes)
6. On the Working Directory enter **$(SolutionDir)** or wherever your **tye.yaml** will be

Set that as your startup project hit F5 and debug. I know it's a hack but for my setup it's working perfectly.

Note that I only tested this using the Basic Example that was provided as I'm not going to use other stuff (e.g. K8s, ingress, replicas, etc.)

I also feel like it made Tye working slower than just doing it via Terminal

**Comment by dominikjeske** [comment]
Are there any official plans for VS support?

**Comment by Joffreyvl** [comment]
@glennfordd did you find a more performant way to run Tye in this way? Or did you make any improvements since your comment?

**Comment by nightroman** [comment]
@Joffreyvl
FWIW, the @glennfordd 's way works very well for me (thank you, Glenn). Like it was said, it's not ideal and may look like a hack but in the end it's just a few ceremonies once, then all works fine.

> Create a project that serves as your Runner or Startup project

I believe, this step is optional. One may configure the existing startup project to run `tye` as required, on F5.
UPDATE: Not always optional/possible, the startup project must use native code debugging.

**Comment by Joffreyvl** [comment]
@nightroman it definitely works, but the child process debugging makes debugging very slow for me.

**Comment by nightroman** [comment]
To slightly improve performance, I use `--no-build` (do not want to debug build processes)

```
tye --no-build ...
```

In this case either build service projects manually before F5 or go to "Tools" / "Options" / "Projects and Solutions" / "Build and Run" and uncheck "Only build startup projects and dependencies on Run".

**Comment by nightroman** [comment]
Another trick that may improve performance, if applicable, set debugging of all other processes (or specified custom) to Managed (.NET Core, .NET 5+), instead of the default "inherit" which presumably includes native debugging:

![image](https://github.com/dotnet/tye/assets/927533/441df897-73c8-457d-b330-ae1306d403fb)
